tests/unittests/test_utils.py

```diff
@@ -32,9 +40,40 @@ def value_factory() -> str:
     assert get_or_add(d, "key2", value_factory) == "value2"
 
 
-def generate_data(value: float, horizon: int, scenarios: int) -> TimeScenarioSeriesData:
+def generate_const_data(
+    value: float, horizon: int, scenarios: int
+) -> TimeScenarioSeriesData:
     data = {}
     for absolute_timestep in range(horizon):
         for scenario in range(scenarios):
             data[TimeScenarioIndex(absolute_timestep, scenario)] = value
     return TimeScenarioSeriesData(time_scenario_series=data)
+
+
+def generate_time_series_data(values: List[float]) -> TimeSeriesData:
+    return TimeSeriesData(
+        time_series={TimeIndex(t): value for t, value in enumerate(values)}
+    )
+
+
+def generate_random_data(
+    mean: float,
+    std: float,
+    horizon: int,
+    scenarios: int,
+    *,
+    seed: Optional[int] = 2024,
+    upper: float = float("inf"),
+    lower: float = float("-inf")
+) -> TimeScenarioSeriesData:
+    X = truncnorm((lower - mean) / std, (upper - mean) / std, loc=mean, scale=std)
+
```

Comment:
All of this is unused

Reply:
I used it when I was testing more general things, but for the UT themselves, I chose a more controlled data generation.
It is much more a utility when developing the idea of the test then the test itself